extension/ackextension/inmemory_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ackextension
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAckPartitionNextAckConcurrency(t *testing.T) {
+	ackSize := 1_000_000
+	ap := newAckStatus(uint64(ackSize))
+	ackIDMap := sync.Map{}
```

Comment:
`sync.Map.Store()` here might help synchronizing order between two goroutines.

Suggest to use regular map for each goroutine. Then merge them and assert len(map)